File: packages/i18nify-js/src/modules/dateTime/__tests__/timezoneToCountry.test.ts
import timezoneToCountry from '../timezoneToCountry';

const mockResponse = {
  metadata_information: {
    IN: {
      timezones: {
        'Asia/Kolkata': { utc_offset: 'UTC +05:30' },
      },
    },
    AG: {
      timezones: {
        'America/Port_of_Spain': { utc_offset: 'UTC -04:00' },
      },
    },
    LC: {
      timezones: {
        'America/Port_of_Spain': { utc_offset: 'UTC -04:00' },
      },
    },
  },
};

beforeEach(() => {
  global.fetch = jest.fn().mockResolvedValue({
    ok: true,
    json: () => Promise.resolve(mockResponse),
  } as unknown as Response);
});

afterEach(() => {
  jest.resetAllMocks();
});

describe('timezoneToCountry', () => {
  it('returns a single country for unique timezones', async () => {
    await expect(timezoneToCountry('Asia/Kolkata')).resolves.toEqual(['IN']);
  });

  it('returns all countries sharing a timezone', async () => {
    const result = await timezoneToCountry('America/Port_of_Spain');
    expect(result).toEqual(['AG', 'LC']);
  });

  it('rejects empty timezone input', async () => {
    await expect(timezoneToCountry('')).rejects.toThrow(
      'timezoneToCountry: timezone must not be empty.',
    );
  });

  it('rejects unknown timezones', async () => {
    await expect(timezoneToCountry('Pacific/Fake')).rejects.toThrow(
      'An error occurred while fetching country codes for timezone "Pacific/Fake"',
    );
  });
});

File: packages/i18nify-go/modules/datetime/timezone_to_country_test.go
package datetime

import "testing"

func TestTimezoneToCountry(t *testing.T) {
	countries, err := TimezoneToCountry("Asia/Kolkata")
	if err != nil {
		t.Fatalf("expected no error, got %v", err)
	}

	if len(countries) != 1 || countries[0] != "IN" {
		t.Fatalf("unexpected countries: %#v", countries)
	}

	countries, err = TimezoneToCountry("America/Port_of_Spain")
	if err != nil {
		t.Fatalf("expected no error, got %v", err)
	}

	if len(countries) != 1 || countries[0] != "TT" {
		t.Fatalf("unexpected countries for America/Port_of_Spain: %#v", countries)
	}

	if _, err := TimezoneToCountry(""); err == nil {
		t.Fatal("expected error for empty timezone")
	}
}
